libs/langchain/langchain/vectorstores/dingo.py

```diff
@@ -0,0 +1,338 @@
+            import dingodb
+    """
+
+    def __init__(
+        self,
+            embedding_function: Callable,
+            text_key: str,
+            client: Any = None,
+            index_name: Optional[str] = None,
+            index_params: Optional[dict] = None,
+            search_params: Optional[dict] = None,
+            host: List[str] = ["172.20.31.10:13000"],
+            user: str = "root",
+            password: str = "123123",
+            self_id: bool =False
+    ):
+        """Initialize with Dingo client."""
+        try:
+            import dingodb
+        except ImportError:
+            raise ValueError(
+                "Could not import dingo python package. "
+                "Please install it with `pip install dingodb."
+            )
+
+         # collection
+        if client is not None:
+            dingo_client = client
+        else:
+            try:
+                # connect to dingo db
+                dingo_client = dingodb.DingoDB(user, password, host)
+            except ValueError as e:
+                raise ValueError(f"Dingo failed to connect: {e}")
+
+        self._text_key = text_key
+        self._client = dingo_client
+
+        if index_name is not None and index_name not in dingo_client.get_index():
+            if self_id is True:
+                dingo_client.create_index(index_name, 1024, auto_id=False)
+            else:
+                dingo_client.create_index(index_name, 1024)
+
+        self._index_name = index_name
+        self._embedding_function = embedding_function
+
+    def add_texts(
+            self,
+            texts: Iterable[str],
+            metadatas: Optional[List[dict]] = None,
+            ids: Optional[List[int]] = None,
+            text_key: str = "text",
+            batch_size: int = 500,
+            **kwargs: Any,
+    ) -> List[str]:
+        """Run more texts through the embeddings and add to the vectorstore.
+
+        Args:
+            texts: Iterable of strings to add to the vectorstore.
+            metadatas: Optional list of metadatas associated with the texts.
+            ids: Optional list of ids to associate with the texts.
+
+        Returns:
+            List of ids from adding the texts into the vectorstore.
+
+        """
+        # Embed and create the documents
+        ids = ids or [int(str(uuid.uuid1().int)[:13]) for _ in texts]
+        metadatas_list = []
+        embeds = []
+        for i, text in enumerate(texts):
+            emb = self._embedding_function(text)
+            metadata = metadatas[i] if metadatas else {}
+            metadata[self._text_key] = text
+            metadatas_list.append(metadata)
+            embeds.append(emb)
+            
+
+        # upsert to Dingo
+        for i in range(0, len(texts), batch_size):
+            j = i + batch_size
+            self._client.vector_add(self._index_name, metadatas_list[i:j], embeds[i:j], ids[i:j])
+
+        return ids
```

Comment:
this is expected to be a list of strings

Reply:
> 

Can I change the return type to List[int]?  Because our incoming id is int type

Reply:
we could use ints for actually uploading but return
```python
return [str(id) for id in ids]
```

Reply:
> we could use ints for actually uploading but return
> 
> ```python
> return [str(id) for id in ids]
> ```

Ok, I try to modify it back to list[str], the python SDK will automatically convert it with pydantic